WebApplication1/WebApplication1/main.js

```diff
@@ -0,0 +1,81 @@
+﻿
+
+var csv = {
+    read: function (location) {
+
+        var user_lat = location.coords.latitude;
+        var user_lon = location.coords.longitude;
+
+        var rawFile = new XMLHttpRequest();
+        rawFile.open("GET", "coffee_shops.csv", true);
+        rawFile.onreadystatechange = function () {
+            if (rawFile.readyState === 4) {
+                if (rawFile.status === 200 || rawFile.status == 0) {
+                    var line = rawFile.responseText.split("\n");
+                    var zzz = [];
+                    for (var i in line) {
+                        var x = line[i].split(",");
+                        if (x.length != 3) {
+                            throw Error("Invalid record in CSV");
+                            break;
+                        }
+
+                        zzz[i] = [];
+                        zzz[i]["name"] = x[0];
+                        zzz[i]["km"] = mats.transformToKM(user_lat, user_lon, x[1], x[2]);
+
+                        zzz.sort(function (a, b) { return a.km - b.km });
```

Comment:
Your sorting this array for each line added, can you come up with a more efficient way?